infoschema/tables.go

```diff
@@ -797,7 +797,7 @@ var tableTiDBIndexesCols = []columnInfo{
 	{name: "SEQ_IN_INDEX", tp: mysql.TypeLonglong, size: 21},
 	{name: "COLUMN_NAME", tp: mysql.TypeVarchar, size: 64},
 	{name: "SUB_PART", tp: mysql.TypeLonglong, size: 21},
-	{name: "INDEX_COMMENT", tp: mysql.TypeVarchar, size: 2048},
+	{name: "INDEX_COMMENT", tp: mysql.TypeVarchar, size: 1024},
```

Comment:
Although the max length of index comment is 1024, it shows that the column length is 2048 in MySQL:

```
mysql> desc information_schema.statistics;
+---------------+---------------+------+-----+---------+-------+
| Field         | Type          | Null | Key | Default | Extra |
+---------------+---------------+------+-----+---------+-------+
| TABLE_CATALOG | varchar(64)   | YES  |     | NULL    |       |
| TABLE_SCHEMA  | varchar(64)   | YES  |     | NULL    |       |
| TABLE_NAME    | varchar(64)   | YES  |     | NULL    |       |
| NON_UNIQUE    | int           | NO   |     | 0       |       |
| INDEX_SCHEMA  | varchar(64)   | YES  |     | NULL    |       |
| INDEX_NAME    | varchar(64)   | YES  |     | NULL    |       |
| SEQ_IN_INDEX  | int unsigned  | NO   |     | NULL    |       |
| COLUMN_NAME   | varchar(64)   | YES  |     | NULL    |       |
| COLLATION     | varchar(1)    | YES  |     | NULL    |       |
| CARDINALITY   | bigint        | YES  |     | NULL    |       |
| SUB_PART      | bigint        | YES  |     | NULL    |       |
| PACKED        | binary(0)     | YES  |     | NULL    |       |
| NULLABLE      | varchar(3)    | NO   |     |         |       |
| INDEX_TYPE    | varchar(11)   | NO   |     |         |       |
| COMMENT       | varchar(8)    | NO   |     |         |       |
| INDEX_COMMENT | varchar(2048) | NO   |     | NULL    |       |
| IS_VISIBLE    | varchar(3)    | NO   |     |         |       |
| EXPRESSION    | longtext      | YES  |     | NULL    |       |
+---------------+---------------+------+-----+---------+-------+
18 rows in set (0.18 sec)
```

Reply:
In MySQL 5.7.36 the length is 1024 instead. And the [reference](https://dev.mysql.com/doc/refman/5.7/en/create-table.html#create-table-indexes-keys) also specifies the length of index comment to 1024

```
mysql> desc information_schema.statistics;
+---------------+---------------+------+-----+---------+-------+
| Field         | Type          | Null | Key | Default | Extra |
+---------------+---------------+------+-----+---------+-------+
| TABLE_CATALOG | varchar(512)  | NO   |     |         |       |
| TABLE_SCHEMA  | varchar(64)   | NO   |     |         |       |
| TABLE_NAME    | varchar(64)   | NO   |     |         |       |
| NON_UNIQUE    | bigint(1)     | NO   |     | 0       |       |
| INDEX_SCHEMA  | varchar(64)   | NO   |     |         |       |
| INDEX_NAME    | varchar(64)   | NO   |     |         |       |
| SEQ_IN_INDEX  | bigint(2)     | NO   |     | 0       |       |
| COLUMN_NAME   | varchar(64)   | NO   |     |         |       |
| COLLATION     | varchar(1)    | YES  |     | NULL    |       |
| CARDINALITY   | bigint(21)    | YES  |     | NULL    |       |
| SUB_PART      | bigint(3)     | YES  |     | NULL    |       |
| PACKED        | varchar(10)   | YES  |     | NULL    |       |
| NULLABLE      | varchar(3)    | NO   |     |         |       |
| INDEX_TYPE    | varchar(16)   | NO   |     |         |       |
| COMMENT       | varchar(16)   | YES  |     | NULL    |       |
| INDEX_COMMENT | varchar(1024) | NO   |     |         |       |
+---------------+---------------+------+-----+---------+-------+
16 rows in set (0.00 sec)
```

Is the length of column in `information_schema.statistics` specified to 2048 in MySQL 8.x ?

Reply:
Looks like 1024 now, https://github.com/mysql/mysql-server/blob/3290a66c89eb1625a7058e0ef732432b6952b435/include/mysql_com.h#L95.